src/stats.ts

```diff
@@ -226,9 +248,27 @@ export abstract class StatsCommon<T extends number | bigint> implements Node.Sta
 			return true;
 		}
 
-		// Mask for
-		const adjusted = (credentials.uid == this.uid ? S_IRWXU : 0) | (credentials.gid == this.gid ? S_IRWXG : 0) | S_IRWXO;
-		return (mode & this.mode & adjusted) == mode;
+		let perm = 0;
+
+		if (credentials.uid === this.uid) {
+			// Owner permissions
+			if (this.mode & S_IRUSR) perm |= R_OK;
+			if (this.mode & S_IWUSR) perm |= W_OK;
+			if (this.mode & S_IXUSR) perm |= X_OK;
+		} else if (credentials.gid === this.gid) {
```

Comment:
Shouldn't this be `if`, not `else if`? For example with permissions 070, if the uid and gid matched this would result in no access, even though the group permissions would permit rwx.